lib/sqlalchemy/sql/functions.py

```diff
@@ -49,7 +55,48 @@ def register_function(identifier, fn, package="_default"):
 
     """
     reg = _registry[package]
-    reg[identifier] = fn
+    case_sensitive_reg = _case_sensitive_registry[package]
+    raw_identifier = identifier
+    identifier = identifier.lower()
+
+    # Check if a function with the same lowercase identifier is registered.
+    if identifier in reg:
```

Comment:
**mike bayer** (zzzeek) wrote:

If we can try the "_CASE_SENSITIVE" symbol idea in [2] below, this line would become:

    if identifier in reg and identifier[reg] is not _CASE_SENSITIVE:

Reply:
I like it!